server/core/dcb.cc

```diff
@@ -878,6 +889,7 @@ dcb_log_errors_SSL(DCB *dcb, int ret)
 int
 dcb_write(DCB *dcb, GWBUF *queue)
 {
+    dcb->writeqlen += gwbuf_length(queue);
```

Comment:
Just a note. We have had situations where calling `gwbuf_length(...)` caused a performance hit. In practice a situation where the GWBUF actually was a very long chain. We need to optimize that case so that the cost is always constant irrespective of how long a chain the GWBUF is.

Reply:
Get every GWBUF's length only once, so I think maybe it is ok; I can test it by   select a huge resultset and very row is small;

Reply:
Yes, this is OK. We created https://jira.mariadb.org/browse/MXS-1704 to remind us that it would be good to optimize `gwbuf_length` at some point.